extension-manifest-v3/src/background/reporting.js

```diff
@@ -216,24 +218,28 @@ const reporting = new Reporting({
   communication,
 });
 
-observe('terms', (terms) => {
-  if (terms) {
-    reporting.init().catch((e) => {
-      console.warn(
-        'Failed to initialize reporting. Leaving the module disabled and continue.',
-        e,
-      );
-    });
-  } else {
-    reporting.unload();
-  }
-});
+const setup = asyncSetup([
+  observe('terms', (terms) => {
```

Comment:
Here, we need to wait for the initialization of reporting (please add async + await):

```
async (terms) => {
  if (terms) {
    await reporting.init().catch(...)
```